frontend/src/test/scala/bloop/bsp/BspClientTest.scala

```diff
@@ -0,0 +1,115 @@
+package bloop.bsp
+
+import java.nio.file.Files
+
+import bloop.cli.Commands
+import bloop.io.AbsolutePath
+import bloop.tasks.ProjectHelpers
+import ch.epfl.`scala`.bsp.schema.{
+  BuildClientCapabilities,
+  InitializeBuildParams,
+  InitializedBuildParams
+}
+import ch.epfl.scala.bsp.endpoints
+import monix.execution.{ExecutionModel, Scheduler}
+import monix.{eval => me}
+import org.langmeta.jsonrpc.{BaseProtocolMessage, Response, Services}
+import org.langmeta.lsp.{LanguageClient, LanguageServer}
+import org.scalasbt.ipcsocket.Win32NamedPipeSocket
+
+import scala.concurrent.duration.FiniteDuration
+
+object BspClientTest {
+  private final val slf4jLogger = com.typesafe.scalalogging.Logger("test")
+
+  def cleanUpLastResources(cmd: Commands.ValidatedBsp) = {
+    cmd match {
+      case cmd: Commands.WindowsLocalBsp => ()
+      case cmd: Commands.UnixLocalBsp =>
+        // We delete the socket file created by the BSP communication
+        if (!Files.exists(cmd.socket)) ()
+        else Files.delete(cmd.socket)
+      case cmd: Commands.TcpBsp => ()
+    }
+  }
+
+  def setupBspCommand(cmd: Commands.ValidatedBsp,
+                      cwd: AbsolutePath,
+                      configDir: AbsolutePath): Commands.ValidatedBsp = {
+    val common = cmd.cliOptions.common.copy(workingDirectory = cwd.syntax)
+    val cliOptions = cmd.cliOptions.copy(configDir = Some(configDir.underlying), common = common)
+    cmd match {
+      case cmd: Commands.WindowsLocalBsp => cmd.copy(cliOptions = cliOptions)
+      case cmd: Commands.UnixLocalBsp => cmd.copy(cliOptions = cliOptions)
+      case cmd: Commands.TcpBsp => cmd.copy(cliOptions = cliOptions)
+    }
+  }
+
+  val scheduler: Scheduler = Scheduler(java.util.concurrent.Executors.newFixedThreadPool(4),
```

Comment:
Why not use the default thread pool? I think that's what we do in other tests

Reply:
Because I want to constrain it by default. I don't want the thread pool for bsp to be reused for the tests because I'm making the tests more restrictive (less threads per pool) to make sure we don't get any race condition. Should be therefore independent.